PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -2717,40 +2717,16 @@ void ScintillaEditView::updateLineNumberWidth()
 {
 	if (_lineNumbersShown)
 	{
-		auto linesVisible = execute(SCI_LINESONSCREEN);
-		if (linesVisible)
+		auto lineCount = execute(SCI_GETLINECOUNT);
+		int numDigits = 0;
+		while (lineCount)

```

Comment:
I would do the following code:
```
		if (lineCount >= 1 && lineCount <= 9) numDigits = 1;
		else if (lineCount >= 10 && lineCount <= 99) numDigits = 2;
		else if (lineCount >= 100 && lineCount <= 999) numDigits = 3;
		else if (lineCount >= 1000 && lineCount <= 9999) numDigits = 4;
		else if (lineCount >= 10000 && lineCount <= 99999) numDigits = 5;
		else if (lineCount >= 100000 && lineCount <= 999999) numDigits = 6;
		else
		{
			while (lineCount)
			{
				lineCount /= 10;
				++numDigits;
			}
		}
```
From 1 to 6 digits will catch 90% of cases - so no loop on scrolling in these cases.

Reply:
That's clever. :+1: 

Just curious: why not `if (lineCount < 10) numDigits = 1;` and so on?

Reply:
Actually we can start with `if (lineCount < 1000) numDigits = 3;` and remove the `max()` later.

Reply:
@Yaron10
> why not if (lineCount < 10) numDigits = 1; and so on?

You're right about it. Just tune up and commited (for the dynamic display part):
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/8e73f5d116b159485bb1b4f95e8f9904efd27632

Reply:
:+1:

Reply:
> Actually we can start with if (lineCount < 1000) numDigits = 3; and remove the max() later.

Good thinking. I guess it's optimized maximum:
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/d82aba7b0ca40470b9cdeef624aa35bdf6376b42

Reply:
Great. :+1: 

I wouldn't have thought of removing `include "tchar.h"`. :)